script/spec-runner.js

```diff
@@ -0,0 +1,81 @@
+const cp = require('child_process')
+const crypto = require('crypto')
+const fs = require('fs')
+const { hashElement } = require('folder-hash')
+const path = require('path')
+
+const BASE = path.resolve(__dirname, '../..')
+const NPM_CMD = process.platform === 'win32' ? 'npm.cmd' : 'npm'
+
+const specHashPath = path.resolve(__dirname, '../spec/.hash')
+if (!fs.existsSync(specHashPath)) {
+  fs.writeFileSync(specHashPath, 'invalid\ninvalid')
+}
+
+const [lastSpecHash, lastSpecInstallHash] = fs.readFileSync(specHashPath, 'utf8').split('\n')
+getSpecHash().then(([currentSpecHash, currentSpecInstallHash]) => {
+  const specChanged = currentSpecHash !== lastSpecHash
+  const installChanged = lastSpecInstallHash !== currentSpecInstallHash
+  if (specChanged || installChanged) {
+    const out = cp.spawnSync(NPM_CMD, ['install'], {
+      env: Object.assign({}, process.env, {
+        npm_config_nodedir: path.resolve(BASE, 'out/Default/gen/node_headers'),
+        npm_config_msvs_version: '2017'
+      }),
+      cwd: path.resolve(__dirname, '../spec')
+    })
+    if (out.status !== 0) {
+      console.error('Failed to npm install in the spec folder')
+      process.exit(1)
+    }
+    return getSpecHash()
+      .then(([newSpecHash, newSpecInstallHash]) => {
+        fs.writeFileSync(specHashPath, `${newSpecHash}\n${newSpecInstallHash}`)
+      })
+  }
+}).then(() => {
+  let exe = path.resolve(BASE, getOutPath())
+  const args = process.argv.slice(2)
+  if (process.platform === 'linux') {
+    args.unshift(path.resolve(__dirname, 'lib/dbus_mock.py'), exe)
+    exe = 'python'
+  }
+  const child = cp.spawn(exe, args, {
+    cwd: path.resolve(__dirname, '../..'),
+    stdio: 'inherit'
+  })
+  child.on('exit', (code) => {
+    process.exit(code)
+  })
+})
+
+function getOutPath () {
```

Comment:
Maybe `getElectronExec()`?

Current name scans as returning the path to `out/`.